comparison_with_mma.tex

```diff
@@ -51,19 +57,117 @@ \subsection{Mathematica}
 Regarding associative expressions,
 SymPy handles associativity by making associative expressions inherit the
 class \texttt{AssocOp},
-while Mathematica specifies the \texttt{Flat} attribute on the expression type.
+while Mathematica specifies the \texttt{Flat}\cite{WolframRefFlat} attribute on the expression type.
+
+% One identity
+
 
 % Pattern matching
 
+Mathematica relies heavily on pattern matching:
+even the so-called equivalent of function declaration is in reality
+the definition of a pattern matching generating an expression tree transformation
+on input expressions.
+%
+Mathematica's pattern matching is sensitive to associative\cite{WolframRefFlat}, commutative\cite{WolframRefOrderless},
+and one-identity\cite{WolframRefOneIdentity} properties of its expression tree nodes\cite{WolframRefFlatAndOrderlessFunctions}.
+%
+SymPy has various ways to perform pattern matching.
+All of them play a lesser role in the CAS than in Mathematica
+and are basically available as a tool to rewrite expressions.
+The differential equation solver in SymPy somewhat relies on pattern matching to
+identify the kind of differential equation, but it is envisaged to replace
+that strategy with analysis of Lie symmetries in the future.
+Mathematica's real advantage is the ability to add new overloading to the
+expression builder at runtime, or for specific subnodes.
+Consider for example
+\begin{verbatim}
+In[1]:= Unprotect[Plus]
```

Comment:
It would be nice to follow one style for displaying code fragments. [This](https://github.com/sympy/sympy-paper/blob/master/solvers.tex#L75) one is mostly being used in the paper.